### DISCUSSION TITLE
Week 10: Aug 19-23

### DISCUSSION THREAD

**Discussion by eeholmes** [SAFS-Varanasi-Internship/Summer-2024#24] [General]
* https://docs.google.com/document/d/1LbIuGtCFFLPO3MWRDquYCZ1IhFUcE8rhn6mDIATLR4Y/edit

**Comment by eeholmes** [comment]
@ShridharS19 Examples

* https://github.com/geo-smart/crunchy-snow
* https://github.com/geo-smart/multi-gpr-soil
* https://github.com/geo-smart/swe_forecasting_prod

**Comment by eeholmes** [comment]
@yifeihang 

I was thinking today about this image.  You are using the Chl-a from the day before and the day after as part of your information used in predicting. I *think* you need to add the green mask to those days if you are not. Otherwise you will have Chl-a observations for most of the green pixels on the day before and day after. That would not be realistic. 

So if you are predicting day 20. 
* Add clouds from day 30 as the green mask.
* Add clouds from day 29 as the green mask for the 'day before' image
* Add clouds rom day 31 as the green mask for the 'day after' image
That way you will pick up a few more of the green pixels from the day before and after (clouds do move) but it won't be excessive, i.e. the fake green cloud was only present for day 20 and then disappeared.

![image](https://github.com/user-attachments/assets/b6f68eec-6af0-4612-9047-cb2ee2344a51)

**Reply by yifeihang** [reply]
Ok will do

**Comment by eeholmes** [comment]
@ShridharS19 A couple things for the hackweek

* Yes, they would like to add your project and to pitch it next week.
* I prepped the project repo: https://github.com/SAFS-Varanasi-Internship/mind-the-chl-gap
* ASAP You need to sign up for CryoCloud https://docs.google.com/forms/d/e/1FAIpQLScb2B5LRy7qU9qDfUp5a51n87sumOxivXbQhc02wFX_FxEbXg/viewform
* Use "Hackweek 2024" as the Award number when you sign up. 
* Get the Slack app on your phone since that will be the main way that people communicate. I will add you to the project_development channel on that.
* Once you are on CryoCloud, test out your notebooks. You should be able to see the GPU option, but if not let me know.

**Comment by eeholmes** [comment]
@ShridharS19 Project lead training. Here is the video from their July 9th training. You can skim most of it but make sure you understand the Day 1 and 2 project formation logistics. I'll be there to help you out with that part but you will do the pitch and act as project lead. I'll act as a support person for the team.

* https://app.slack.com/client/T070R5BPMNY/C07217M12P8

**Comment by eeholmes** [comment]
@yifeihang I sent you an invite for the geo-smart hackweek repo. https://github.com/geo-smart/mind-the-chl-gap. We are working on notebooks describing the gap-filling problem and various machine-learning models. Feel free to follow along and contribute.

Adding or polishing off the tree models for gap-filling might be a  good project for OceanHackWeek to build on this Jupyter Book since I don't think you'll have GPU for OceanHackWeek.  Today there was a presentation on XGBoost.
* https://docs.google.com/presentation/d/1ncn8rEeRp0bPiE45omQSZDZM17cUFjIE/edit#slide=id.p1
* https://xgboost.readthedocs.io/en/stable/
* tutorials: https://github.com/geo-smart/website-2024/tree/main/book/tutorials/decision_trees

**Comment by eeholmes** [comment]
@yifeihang @ShridharS19 Trying to run CHL_prediction_CNN.ipynb but it says there is not tensorflow.keras module. I am on https://fish.opensci.live/ with the GPU and tensorflow image. Do you know what I am doing wrong?

<img width="859" alt="image" src="https://github.com/user-attachments/assets/17f2fc33-018c-46a6-b4a0-072da26da545">

**Reply by yifeihang** [reply]
Maybe try directly using ”keras” instead of “tensorflow.keras”?

**Reply by eeholmes** [reply]
That worked!

**Reply by eeholmes** [reply]
@yifeihang When you have a chance, can you upload `"E:\24SU Varanasi Intern\参考资料\2023_Intern_Material\land_mask_nc.npy"` to https://github.com/SAFS-Varanasi-Internship/yifei_repo_2024 ?  We need it to run the CNN tutorial. Thanks!

**Reply by yifeihang** [reply]
Just uploaded it, and it's under shared\data

**Comment by eeholmes** [comment]
Thanks! If you have any other tutorials, can you upload those too? Looking
for gap-filling examples. The CNN tutorial is for filling in a missing day
rather than filling gaps (rather than missing pixels).

**Reply by yifeihang** [reply]
I just uploaded the gapfilling version to my github repo. I've made a few changes regarding the previous & next day CHL's masking problem in another notebook, but I haven't updated this tutorial. Will do that later today or latest tomorrow.
